recipe/meta.yaml

```diff
@@ -49,29 +49,27 @@ requirements:
   run:
     - python
     - packaging
+    - setuptools >=40.6.0
     - {{ pin_compatible('pyarrow', max_pin='x') }}
-    # While requests is vendored, we use regular requests to perform OCSP checks
-    - requests <3.0.0
-    - pytz
-    - pycryptodomex >=3.2,!=3.5.0,<4.0.0
-    - pyOpenSSL >=16.2.0,<23.0.0
+    - asn1crypto >0.24.0,<2.0.0
     - cffi >=1.9,<2.0.0
-    - cryptography >=3.1.0,<39.0.0
-    - pyjwt <3.0.0
+    - cryptography >=3.1.0,<42.0.0
     - oscrypto <2.0.0
-    - asn1crypto >0.24.0,<2.0.0
-    # A functioning pkg_resources.working_set.by_key and pkg_resources.Requirement is
-    # required. Python 3.6 was released at the end of 2016. setuptools 34.0.0 was released
-    # in early 2017, so we pick this version as a reasonably modern base.
-    - setuptools >34.0.0
-    # requests requirements
+    - pyOpenSSL >=16.2.0,<24.0.0
+    - pycryptodomex >=3.2,!=3.5.0,<4.0.0
+    - pyjwt <3.0.0
+    - pytz
+    - requests <3.0.0
+    - packaging
     - charset-normalizer >=2,<4
     - idna >=2.5,<4
     - urllib3 >=1.21.1,<1.27
     - certifi >=2017.4.17
     - typing_extensions >=4.3,<5
     - filelock >=3.5,<4
     - sortedcontainers >=2.4.0
+    - platformdirs >=2.6.0,<3.9.0
+    - tomlkit
   run_constrained:
```

Comment:
Upstream deps for review:
![image](https://github.com/conda-forge/snowflake-connector-python-feedstock/assets/3316177/c3795f0a-9d1e-4765-b206-813b73193996)